src/components/tickets/index.tsx

```diff
@@ -0,0 +1,165 @@
+        : '';
+
+    return apiUrl;
+  };
+
+  const fetchTickets = async () => {
+    try {
+      const apiUrl = buildApiUrl();
+      const response = await TicketService.get(apiUrl);
+      setTicketsData(response.data);
+    } catch (error) {
+      throw new Error('Failed to fetch tickets');
+    }
+  };
+
+  const toggleModal = useCallback(() => {
+    setIsModalOpen((prev) => !prev);
+  }, []);
+
+  const toggleUpdateModal = useCallback((id?: string) => {
+    setIsUpdateModalOpen((prev) => !prev);
+    id && setParamsId(id);
+  }, []);
+
+  const handleEditClick: SubmitHandler<IUpdateTickets> = async (data: IUpdateTickets) => {
+    const body = { ...data };
+    await dispatch(updateTicketAsync({ id: paramsId, data: body }));
+    fetchTickets();
+    toggleUpdateModal();
+  };
+
+  const handleDeleteClick = async (rowId: string) => {
+    await dispatch(deleteTicketAsync(rowId));
+    fetchTickets();
+  };
+
+  return (
+    <StyledBox>
+      {ticketsData && (
+        <StyledDataGrid
+          autoHeight
+          rows={ticketsData}
+          rowHeight={92}
+          columns={[
+            ...columns,
+            {
+              field: 'menu',
+              headerName: '',
+              width: 15,
+              sortable: false,
+              renderCell: (params) => {
+                return (
+                  <>
+                    <Stack>
+                      <CustomSelect IconComponent={MoreVertIcon}>
+                        <StyledMenuItem onClick={() => toggleUpdateModal(params.row.id)}>
+                          Edit
+                        </StyledMenuItem>
+                        <StyledMenuItem onClick={() => handleDeleteClick(params.row.id)}>
+                          Delete
+                        </StyledMenuItem>
+                      </CustomSelect>
+                    </Stack>
+                  </>
+                );
+              },
+            },
+          ]}
+          pageSizeOptions={pageSizeOptions}
+          paginationModel={paginationModel}
+          onPaginationModelChange={setPaginationModel}
+          disableRowSelectionOnClick
+          disableColumnMenu
+          slots={{
+            toolbar: () => (
+              <CustomToolbar
+                dispatch={dispatch}
+                fetchTickets={fetchTickets}
+                toggleModal={toggleModal}
+                setSearchParams={setSearchParams}
+                selectedPriorities={selectedPriorities}
+                setSelectedPriorities={setSelectedPriorities}
+                isModalOpen={isModalOpen}
+              />
+            ),
+          }}
+        />
+      )}
+      {isUpdateModalOpen && (
+        <>
+          <TicketsModal
+            toggleModal={toggleUpdateModal}
+            handleForm={handleEditClick}
+            refetchTickets={fetchTickets}
+            initialValues={ticketsData?.find((ticket) => ticket.id === paramsId) || null}
```

Comment:
First of all we don't use `handleEditClick` except of this modal, we don't need it here.  There are two ways for now 1) do get request by id in the modal and use the response.2) Better to create state like `selectedTicket` and then pass it through the props

Reply:
fix: handleEditClick and also handleCreateTicket moved in TicketsForm.  selectedTicket state created and then passed as the props